DylibBinder/StreamWriterExtensions.cs

```diff
@@ -0,0 +1,887 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Dynamo.SwiftLang;
+using SwiftReflector.Inventory;
+using SwiftReflector.SwiftXmlReflection;
+using SwiftReflector.TypeMapping;
+
+namespace DylibBinder {
+	public static class StreamWriterExtensions {
+
+		static int IndentLevel { get; set; }
+		static int privateIndex { get; set; } = 0;
+
+		public static void WriteXmlFile (this StreamWriter sw, string moduleName, ModuleInventory mi)
+		{
+			(var classesList, var structsList, var enumsList) = CheckInventory.GetClassesStructsEnums (mi);
+			var protocolsList = CheckInventory.GetProtocols (mi);
+
+			Console.WriteLine ($"Extracting \"{moduleName}\"");
+
+			var innerX = new InnerX ();
+			var innerXDictionary = innerX.AddClassContentsList (classesList, enumsList, structsList);
+			IndentLevel = 1;
+
+			sw.WriteModuleIntro (moduleName);
+			sw.WriteClassContentsList (classesList, "class", innerXDictionary, 0);
+			sw.WriteClassContentsList (enumsList, "enum", innerXDictionary, 0);
+			sw.WriteClassContentsList (structsList, "struct", innerXDictionary, 0);
+			sw.WriteProtocols (moduleName, protocolsList);
+			sw.WriteModuleOutro ();
+		}
+
+		public static void WriteXmlIntro (this StreamWriter sw, string version)
+		{
+			sw.WriteLineWithIndent ($"<?xml version=\"{version}\" encoding=\"utf-8\"?>");
+			sw.WriteLineWithIndent ($"<xamreflect version=\"{version}\">");
+			Indent ();
+			sw.WriteLineWithIndent ($"<modulelist>");
+		}
```

Comment:
No. Sorry, you are using the wrong approach here. You are re-implementing the XmlWriter class.  All this methods son be re-written using the XmlWriter class.